tests/unit_tests/experiment_plans/test_multi_rotation_scan_plan.py

```diff
@@ -0,0 +1,368 @@
+                chi := written_nexus_file["/entry/sample/sample_chi/chi"], h5py.Dataset
+            )
+            assert chi[:] == scan.chi_start_deg
+            assert isinstance(
+                phi := written_nexus_file["/entry/sample/sample_phi/phi"], h5py.Dataset
+            )
+            assert phi[:] == scan.phi_start_deg
+            assert isinstance(
+                omega := written_nexus_file["/entry/sample/sample_omega/omega"],
+                h5py.Dataset,
+            )
+            omega = omega[:]
+            assert isinstance(
+                omega_end := written_nexus_file["/entry/sample/sample_omega/omega_end"],
+                h5py.Dataset,
+            )
+            omega_end = omega_end[:]
+            assert len(omega) == scan.num_images
+            expected_omega_starts = np.linspace(
+                scan.omega_start_deg,
+                scan.omega_start_deg
+                + ((scan.num_images - 1) * multi_params.rotation_increment_deg),
+                scan.num_images,
+            )
+            assert np.allclose(omega, expected_omega_starts)
+            expected_omega_ends = (
+                expected_omega_starts + multi_params.rotation_increment_deg
+            )
+            assert np.allclose(omega_end, expected_omega_ends)
+            assert isinstance(
+                omega_transform := written_nexus_file[
+                    "/entry/sample/transformations/omega"
+                ],
+                h5py.Dataset,
+            )
+            assert isinstance(omega_vec := omega_transform.attrs["vector"], np.ndarray)
+            assert tuple(omega_vec) == (1.0 * scan.rotation_direction.multiplier, 0, 0)
+
+
+@patch("hyperion.external_interaction.callbacks.rotation.ispyb_callback.StoreInIspyb")
+def test_full_multi_rotation_plan_ispyb_called_correctly(
+    mock_ispyb_store: MagicMock,
+    RE: RunEngine,
+    test_multi_rotation_params: MultiRotationScan,
+    fake_create_rotation_devices: RotationScanComposite,
+):
+    callback = RotationISPyBCallback()
+    mock_ispyb_store.return_value = MagicMock(spec=StoreInIspyb)
+    _run_multi_rotation_plan(
+        RE, test_multi_rotation_params, fake_create_rotation_devices, [callback]
+    )
+    ispyb_calls = mock_ispyb_store.call_args_list
+    for instantiation_call, ispyb_store_calls, rotation_params in zip(
+        ispyb_calls,
+        [  # there should be 4 calls to the IspybStore per run
+            mock_ispyb_store.return_value.method_calls[i * 4 : (i + 1) * 4]
+            for i in range(len(test_multi_rotation_params.rotation_scans))
+        ],
+        test_multi_rotation_params.single_rotation_scans,
+    ):
+        assert instantiation_call.args[0] == CONST.SIM.ISPYB_CONFIG
+        assert ispyb_store_calls[0][0] == "begin_deposition"
+        assert ispyb_store_calls[1][0] == "update_deposition"
+        assert ispyb_store_calls[2][0] == "update_deposition"
+        assert ispyb_store_calls[3][0] == "end_deposition"
```

Comment:
I think if you used the `mock_ispyb_conn` fixture (you might have to tweak it to generate more than 2 ids) and the real ispyb store instead of a mock, you could actually assert the contents of the rows sent to ispyb. Then between this and the other tests you would pretty much have an almost full system test of the whole thing, but in the unit tests which would mean we would notice if it breaks!